### ISSUE TITLE
Fletch default help-page is unhelpful.

### ISSUE THREAD

**Issue by sigurdm** [dart-lang/fletch#104]
```
$ out/DebugIA32Clang/fletch 
   help      Display this information.
             Use 'fletch help all' for a list of all actions

   show      List things or show information about a thing
```
As I understand it `show` does not work at all - and the description is too general to really understand what it is supposed to do.

**Comment by madsager** [comment]
I agree, I think we should remove the show verb for now. It is unimplemented and as far as I can tell nobody is working on it.

**Comment by peter-ahe-google** [comment]
I agree, I want to get rid of all prototypes.

**Comment by ErikCorryGoogle** [comment]
I would support just printing the output of help all instead of requiring an extra command.

**Comment by peter-ahe-google** [comment]
I think it is important that we limit the initial help message, but hopefully @lukechurch can investigate what is more helpful in user studies.

**Comment by peter-ahe-google** [comment]
There's precedence both from git and gcc, in limiting the amount of help text shown in the initial help page.

**Comment by ErikCorryGoogle** [comment]
They both limit it to about the size of the output of "fletch help all"

erikcorry@bbc:~/fletch-repo/fletch$ ./out/ReleaseIA32/fletch help all | wc
     90     303    2415
erikcorry@bbc:~/fletch-repo/fletch$ gcc --help | wc
     60     428    3905
erikcorry@bbc:~/fletch-repo/fletch$ git help | wc
     42     282    2050


On 7 September 2015 at 13:55, Peter von der Ahé <notifications@github.com>
wrote:

> There's precedence both from git and gcc, in limiting the amount of help
> text shown in the initial help page.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dart-lang/fletch/issues/104#issuecomment-138278694>.
>



-- 
Erik Corry

Google UK Limited Registered Office: Belgrave House, 76 Buckingham Palace
Road, London SW1W 9TQ Registered in England Number: 3977902

**Comment by peter-ahe-google** [comment]
`show` verb removed in [CL 1346223003](https://codereview.chromium.org/1346223003/).

**Comment by peter-ahe-google** [comment]
@ErikCorryGoogle the current size of the output of `fletch help all` isn't complete and I expect that it will grow substantially over the next few months. Regardless, it is already twice as big as the output of `git --help`. Between versions 2.2.1 and 2.6.0.rc0..., the output from `git --help` has grown by 30%. I consider this an issue as it doesn't scale, eventually you dump so much information on the user that they don't know where to start. Perhaps 

I want the default help message to be concise and fit within a default terminal window, because I don't want to flood the user with information that they don't need.

That being said, the current text is really bad.

**Comment by peter-ahe-google** [comment]
Proposed fix in [CL 1346313003](https://codereview.chromium.org/1346313003/)
